staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go

```diff
@@ -68,17 +70,26 @@ type QueueSet interface {
 type QueueSetConfig struct {
 	// Name is used to identify a queue set, allowing for descriptive information about its intended use
 	Name string
+
 	// ConcurrencyLimit is the maximum number of requests of this QueueSet that may be executing at a time
 	ConcurrencyLimit int
+
 	// DesiredNumQueues is the number of queues that the API says
 	// should exist now.  This may be zero, in which case
 	// QueueLengthLimit, HandSize, and RequestWaitLimit are ignored.
 	DesiredNumQueues int
+
 	// QueueLengthLimit is the maximum number of requests that may be waiting in a given queue at a time
 	QueueLengthLimit int
-	// HandSize is a parameter of shuffle sharding.  Upon arrival of a request, a queue is chosen by randomly
-	// dealing a "hand" of this many queues and then picking one of minimum length.
-	HandSize int
+
+	// Dealer does the shuffle sharding.  When it is time to enqueue a
+	// request, the Dealer is called to deal a random "hand" of queue
+	// indices and the request is put in one of the dealt queues with
+	// minimum length.  The Dealer should deal valid queue indices.
+	// If the Dealer is nil or it deals no valid queue indices then a
```

Comment:
I would prefer to crash rather than make terrible choices? Especially when this being nil or not should be under our control?

Reply:
done